src/backend/distributed/worker/worker_drop_protocol.c

```diff
@@ -142,6 +143,76 @@ worker_drop_distributed_table(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * worker_drop_shell_table drops the shell table of with the given distributed
+ * table without deleting related entries on pg_dist_placement, pg_dist_shard
+ * and pg_dist_placement. We've separated that logic since we handle object
+ * dependencies and table metadata separately while activating nodes.
+ */
+Datum
+worker_drop_shell_table(PG_FUNCTION_ARGS)
```

Comment:
Two important notes:

1) It seems like `DROP TABE` uses `worker_drop_distributed_table` and it does not drop `pg_dist_object` entry. We should fix it and add a test.

```SQL
set citus.grep_remote_commands TO '%worker_drop%';
drop table part_table_1;
NOTICE:  issuing SELECT worker_drop_distributed_table('public.part_table_1')
DETAIL:  on server onderkalaci@localhost:9700 connectionId: 1
....
```

On the worker, we have undefined object, which is the table we just dropped:
```SQL
SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object;
  pg_identify_object_as_address   
----------------------------------
 (table,"{public,part_table}",{})
 (role,{onderkalaci},{})
 (database,{postgres},{})
 (schema,{public},{})
 (relation,,)
(5 rows)

Time: 0.518 ms
```

2) I keep commenting on this UDF :) As a high level note, with the addition of `worker_drop_shell_table` we are creating a dependency between `activate_node` and `worker_drop_shell_table`. In other words, the latter is useless (even dangerous) to be called alone. The problem is that we'd be dropping the table but the metadata would stay.

Given that UDFs can be called by users freely, I think we at a minimum need some sort of additional check, like `ErrorIfMetadataExists` or such here such that users cannot call this UDF for existing distributed tables.

The `ErrorIfMetadataExists` should make sure that there are no entries in pg_dist_partition and pg_dist_object. Tis saves us the future callers of this UDF would not shoot themselves.

Plus, it in a sense makes the dependency between `activate_node` and this UDF explicit.

Reply:
Check not on coordinator.